include/cutlass/gemm/kernel/intel_pvc_gemm.hpp

```diff
@@ -171,33 +180,32 @@ class GemmUniversal<
     return Status::kSuccess;
   }
 
-  static dim3
-  get_grid_shape(Params const& params) {
-    int batch_count = 1;
-    if constexpr (cute::rank(ProblemShape{}) == 4) {
-      batch_count = cute::size<3>(params.problem_shape);
-    }
+  static dim3 get_grid_shape(Params const &params) {
+    auto M = get<0>(params.problem_shape);
+    auto N = get<1>(params.problem_shape);
+    auto L = get<3>(params.problem_shape);
 
-    return dim3(
-            cute::size(cute::ceil_div(cute::shape<0>(params.problem_shape), cute::shape<0>(WorkgroupTileShape{}))),
-            cute::size(cute::ceil_div(cute::shape<1>(params.problem_shape), cute::shape<1>(WorkgroupTileShape{}))),
-            batch_count
-    );
+    int const sg_m = cute::ceil_div(M,
+        CollectiveMainloop::wg_tile_m); // sub_groups required to
+                                        // process A fragments
+    int const sg_n = cute::ceil_div(N,
+        CollectiveMainloop::wg_tile_n); // sub_groups required to
+                                        // process B fragments
+
+    return dim3(sg_n, sg_m, L);
   }
 
-  static dim3
-  get_block_shape() {
-    return dim3(MaxThreadsPerBlock, 1, 1);
+  static dim3 get_block_shape() {
+    return dim3(
+        cute::ceil_div(CollectiveMainloop::wg_tile_n, CollectiveMainloop::sg_tile_n / SubgroupSize),
+        cute::ceil_div(CollectiveMainloop::wg_tile_m, CollectiveMainloop::sg_tile_m), 1);
   }
```

Comment:
All these changes are going to break how cute work internally, so we won't be able to use cute to automatically calculate the tensor layout that should be use for each copy and mma operation. Why are these changes needed.

Reply:
Prefetch needs to know the struct of thread space, but it's ok to get the information from wg_tile and sg_tile. So I reverted here.